ios/Classes/TiNfcVTagTechnology.m

```diff
@@ -9,4 +9,701 @@
+                                                        @"responseDataLength" : responseData.length
+                                                      }];
+                                               }];
+}
+
+- (void)extendedReadSingleBlockWithRequestFlags:(id)args
+{
+  NSNumber *requestFlagsValue = [[args firstObject] valueForKey:@"requestFlags"];
+  uint8_t requestFlags = [requestFlagsValue unsignedCharValue];
+
+  NSNumber *blockNumber = [[args firstObject] valueForKey:@"blockNumber"];
+
+  [[self.tagProxy asNFCISO15693Tag] extendedReadSingleBlockWithRequestFlags:requestFlags
+                                                                blockNumber:[blockNumber integerValue]
+                                                          completionHandler:^(NSData *_Nonnull data, NSError *_Nullable error) {
+                                                            if (![self _hasListeners:@"didExtendedReadSingleBlockWithRequestFlags"]) {
+                                                              return;
+                                                            }
+                                                            TiBuffer *responseData = [[TiBuffer alloc] _initWithPageContext:[self pageContext]];
+                                                            NSMutableData *responsevalue = [NSMutableData dataWithData:data];
+                                                            [responseData setData:responsevalue];
+                                                            [self fireEvent:@"didExtendedReadSingleBlockWithRequestFlags"
+                                                                 withObject:@{
+                                                                   @"errorCode" : error != nil ? NUMINTEGER([error code]) : [NSNull null],
+                                                                   @"errorDescription" : error != nil ? [error localizedDescription] : [NSNull null],
+                                                                   @"errorDomain" : error != nil ? [error domain] : [NSNull null],
+                                                                   @"responseDataLength" : responseData.length
+                                                                 }];
+                                                          }];
+}
+
+- (void)extendedWriteSingleBlockWithRequestFlags:(id)args
+{
+  NSNumber *requestFlagsValue = [[args firstObject] valueForKey:@"requestFlags"];
+  uint8_t requestFlags = [requestFlagsValue unsignedCharValue];
+
+  NSNumber *blockNumber = [[args firstObject] valueForKey:@"blockNumber"];
+
+  NSArray *dataBlock = [[args firstObject] valueForKey:@"dataBlock"];
+  unsigned int dataValue[dataBlock.count];
+  for (int i = 0; i < dataBlock.count; i++) {
+    dataValue[i] = [dataBlock[i] unsignedIntValue];
+  }
+  NSData *data = [[NSData alloc] initWithBytes:dataValue length:dataBlock.count];
```

Comment:
Same here, since it is repetitive code block, better to create a separate methods with name like **arrayToData**.

Reply:
Done